### ISSUE TITLE
Path and locations info improvement when type error returned

### ISSUE THREAD

**Issue by ph55** [prisma/graphql-middleware#96]
Using this library for permission management.
But there is some misleading info in path when validation of `type` failed.

Example.

Schema:
```gql
type Query {
  user(id: Int!): User
}

type User {
  id: Int!
  email: String!
  username: String!
}
```

Apply middleware on `User` type:
```ts
schema = applyMiddleware(schema, {
  User: () => new ForbiddenError(`Access denied`)
})
```

then following query (that return `User` type)
```gql
{
  user(id: 1) {
    username
    email
  }
}
```

throwing error:
```json
{
  "errors": [
    {
      "message": "Access denied",
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "path": [
        "user",
        "username"
      ],
      "extensions": {
        "code": "FORBIDDEN"
      }
    }
  ],
  "data": {
    "user": null
  }
}
```

Look at `locations` and `path` - it includes first queried attribute of `User` (`username`). Which is misleading.
Ideally it should show information regarding `user` query only:
```json
"locations": [
  {
    "line": 2,
    "column": 3
  }
],
"path": [
  "user"
]
```

---

Basically following middlewares currently return same `path` and `location` in error when they shouldn't:
```ts
[
  {
    User: () => new ForbiddenError(`Access denied`)
  },
  {
     User: {
       username: () => new ForbiddenError(`Access denied`)
     }
   }
]
```
